### PR TITLE
Add support for readonly and class properties

### PR DESCRIPTION
**Pull request by brianquinlan**
Fixes https://github.com/dart-lang/native/issues/296

### CODE REVIEW

File: lib/src/code_generator/objc_interface.dart [hunk 1]
```diff
@@ -209,35 +221,56 @@ class ObjCInterface extends BindingType {
...
+          case ObjCMethodKind.propertyGetter:
+            // static returnType getMethodName(NativeLibrary _lib)
+            s.write(_getConvertedType(returnType, w, name));
+            s.write(' get');
+            s.write(methodName[0].toUpperCase() + methodName.substring(1));
+            s.write(paramsToString([], isStatic: true));
+            break;
+          case ObjCMethodKind.propertySetter:
+            // static void setMethodName(NativeLibrary _lib, ...)
+            s.write(' void set');
```
Comment (liamappelbe) [inline lib/src/code_generator/objc_interface.dart:245]: Nit: This setter should have a `returnType` of void, so you can change this to be consistent with this other cases, and that means you can pull the `s.write(_getConvertedType(returnType, w, name));` out the front where the static keyword is.

Similarly, the `paramsToString` call can be the same for all 3 cases, so can be moved below the switch.

File: lib/src/code_generator/objc_interface.dart [hunk 2]
```diff
@@ -247,8 +280,12 @@ class ObjCInterface extends BindingType {
       }
       s.write('    $selName ??= '
           '${_builtInFunctions.registerName}(_lib, "${m.originalName}");\n');
-      final convertReturn = _needsConverting(returnType);
-      s.write('    ${convertReturn ? 'final _ret = ' : 'return '}');
+      final convertReturn = m.kind != ObjCMethodKind.propertySetter &&
```
Comment (liamappelbe) [inline lib/src/code_generator/objc_interface.dart:283]: Maybe this check should be whether the return type is void, rather than checking if its a setter?

**Review by liamappelbe — APPROVED** [review]
LGTM, with a couple of nits

**Comment by brianquinlan** [comment]
@liamappelbe I don't have permission to merge, could you press the button for me?

**Comment by dcharkes** [comment]
@brianquinlan I've given you merge permissions.

**Pull request merged**